test/JobManager.spec.js

```diff
@@ -1,37 +1,115 @@
 const JobManager = artifacts.require('./JobManager.sol');
+const { assertFail } = require('./utils.js');
 require('truffle-test-utils').init();
 
 contract('JobManager', accounts => {
 
   let jm;
+  const MINERAL_COMPUTE = 1;
+  const MINEARL_STORAGE = 2;
 
   describe('submitMineral', () => {
 
     before(async () => {
       jm = await JobManager.deployed();
     });
 
+    it('should fail if category is not MINERAL_COMPUTE or MINERAL_STORAGE', async () => {
+      await assertFail( jm.submitMineral("test", 0) );
+      await jm.submitMineral("test", MINERAL_COMPUTE);
+      await jm.submitMineral("test", MINEARL_STORAGE);
+      await assertFail( jm.submitMineral("test", 3) );
+    });
+
     it('should store the Mineral in the minerals mapping', async () => {
-      await jm.submitMineral('multiplication');
-      const mineral = await jm.minerals.call(0);
-      assert.equal('multiplication', mineral);
+      const mineralId = (await jm.numberOfMinerals.call()).toNumber();
+      await jm.submitMineral('multiplication', MINERAL_COMPUTE);
+      const mineral = await jm.minerals.call(mineralId);
+      let [name, category] = mineral;
+      assert.equal('multiplication', name);
+      assert.equal(MINERAL_COMPUTE, category);
     });
 
-    it('should increase the value of numberOfMinerals', async () => {
+    it('should increment numberOfMinerals', async () => {
       const numberOfMinerals = await jm.numberOfMinerals.call();
-      await jm.submitMineral('addition');
+      await jm.submitMineral('addition', MINERAL_COMPUTE);
       assert.deepEqual(numberOfMinerals.add(1), await jm.numberOfMinerals.call())
     });
 
     it('should emit a MineralAdded event', async () => {
-      let result = await jm.submitMineral('division');
+      const mineralId = (await jm.numberOfMinerals.call()).toNumber();
+      let result = await jm.submitMineral('division', MINERAL_COMPUTE);
       assert.web3Event(result, {
         event: 'MineralAdded',
         args: {
-          id: 2,
+          id: mineralId,
           name: 'division',
+          category: MINERAL_COMPUTE,
+        }
+      });
+    });
+
+    it('should accept no name for Mineral', async () => {
+      const mineralId = (await jm.numberOfMinerals.call()).toNumber();
+      await jm.submitMineral("", MINERAL_COMPUTE);
```

Comment:
We should probably be associating minerals with the users who submitted them and requiring a non-empty string for a name. If I had created a few minerals, how I would I be able to search them with this architecture?